### PR TITLE
Add integration test for resource-manager garbage collector

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area testing dev-productivity
/kind test
/merge squash

**What this PR does / why we need it**:
Add integration test for resource-manager garbage collector.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/4805

**Release note**:

```breaking dependency
NONE
```

### CODE REVIEW

File: test/integration/resourcemanager/garbagecollector/garbagecollector_test.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+			secretList := &corev1.SecretList{}
+			g.Expect(testClient.List(ctx, secretList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return secretList.Items
+		}).Should(onlyContainObject(withName("secret6")))
+
+		Eventually(func(g Gomega) []corev1.ConfigMap {
+			configMapList := &corev1.ConfigMapList{}
+			g.Expect(testClient.List(ctx, configMapList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return configMapList.Items
+		}).Should(onlyContainObject(withName("configmap0")))
```
Comment (ialidzhikov) [inline test/integration/resourcemanager/garbagecollector/garbagecollector_test.go]: I think here you rather want to assert that this ConfigMap will be garbage collected and will be no longer present in the list. Right now the assertion verifies that the Configmap is present in the list. And I think this currently contradicts with the test goal (feel free to correct me if I miss something).

Probably something like?
```suggestion
		}).Should(Not(ContainElement(withName("configmap0"))))
```

File: test/integration/resourcemanager/garbagecollector/garbagecollector_test.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+		for _, obj := range garbageCollectableObjects {
+			Expect(testClient.Create(ctx, obj)).To(Succeed())
+		}
+
+		Eventually(func(g Gomega) []corev1.Secret {
+			secretList := &corev1.SecretList{}
+			g.Expect(testClient.List(ctx, secretList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return secretList.Items
+		}).Should(onlyContainObject(withName("secret6")))
```
Comment (ialidzhikov) [inline test/integration/resourcemanager/garbagecollector/garbagecollector_test.go]:
```suggestion
		}).Should(Not(ContainElement(withName("secret6"))))
```

File: test/integration/resourcemanager/garbagecollector/garbagecollector_test.go (hunk 1 above)
Comment (rfranzke) [reply test/integration/resourcemanager/garbagecollector/garbagecollector_test.go]: Yes, indeed, nice catch, thank you very much. I made it more clear now based on your suggestion!

File: test/integration/resourcemanager/garbagecollector/garbagecollector_test.go [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+			g.Expect(testClient.List(ctx, configMapList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return configMapList.Items
+		}).Should(And(
+			Not(ContainElement(withName("configmap0"))),
+			ContainElement(withName("configmap1")),
+			ContainElement(withName("configmap2")),
+			ContainElement(withName("configmap3")),
+			ContainElement(withName("configmap4")),
+			ContainElement(withName("configmap5")),
+			ContainElement(withName("configmap6")),
```
Comment (ialidzhikov) [inline test/integration/resourcemanager/garbagecollector/garbagecollector_test.go]:
```suggestion
			ContainElements(withName("configmap1"), withName("configmap2"), withName("configmap3"), withName("configmap4"), withName("configmap5"), withName("configmap6")),
```

File: test/integration/resourcemanager/garbagecollector/garbagecollector_test.go [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+			secretList := &corev1.SecretList{}
+			g.Expect(testClient.List(ctx, secretList, client.InNamespace(testNamespace.Name), client.MatchingLabels(testLabels))).To(Succeed())
+			return secretList.Items
+		}).Should(And(
+			ContainElement(withName("secret0")),
+			ContainElement(withName("secret1")),
+			ContainElement(withName("secret2")),
+			ContainElement(withName("secret3")),
+			ContainElement(withName("secret4")),
+			ContainElement(withName("secret5")),
```
Comment (ialidzhikov) [inline test/integration/resourcemanager/garbagecollector/garbagecollector_test.go]:
```suggestion
			ContainElements(withName("secret0"), withName("secret1"), withName("secret2"), withName("secret3"), withName("secret4"), withName("secret5")),
```

**Review by ialidzhikov — COMMENTED** [review]
Nit, otherwise lgtm

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**